### ISSUE TITLE
Turn on `onnx.checker.check_model` in `OrtEvaluator`

### ISSUE THREAD

**Issue by BowenBao** [microsoft/onnx-script#423]
Right now it's disabled. 
https://github.com/microsoft/onnx-script/blob/de7b00a57894d5cbbfb8a739c5a49f80f73a34cd/onnxscript/evaluator.py#L188

An example script of erroneous onnx model that would be otherwise detected.

```python
import onnxscript
import onnx
import numpy as np

from onnxscript import opset17 as op


@onnxscript.script()
def some_conditional(
    x: onnxscript.FLOAT[2], y: onnxscript.FLOAT[2], dtype: int = -1
) -> onnxscript.FLOAT[2]:
    if dtype != -1:
        tmp = op.Cast(x, to=dtype)
    else:
        tmp = x

    x = op.CastLike(tmp, x)
    return x + y


x = np.array([1.0, 2.0])
# Passed
print(some_conditional(x, x, -1))
print(some_conditional(x, x, 7))

model_proto = some_conditional.to_model_proto()
# Passed
onnx.checker.check_model(model_proto, full_check=False)
# Failed
onnx.checker.check_model(model_proto, full_check=True)

```

**Comment by gramalingam** [comment]
Thanks for creating this issue. Agree that we should enable that check (there was some issue earlier). It will help catch some issues early on, but it won't catch this particular issue ... the reason is that in eager-mode, for control-flow constructs we rely on python execution (for a better debugging experience) and don't invoke ONNX If.

The right place to catch this is when creating a ModelProto. We could try to catch it at script-time, but there's a challenge ... we need some extra context information (providing types for inputs, and attribute-values) to do a check of a Function.

**Comment by BowenBao** [comment]
You are right, I see that current location of checker isn't for the whole function.

Could we invoke checker at `Evaluator::eval_function`? It has all the info (function and args).

**Comment by BowenBao** [comment]
More discussions in this thread https://github.com/microsoft/onnx-script/pull/421#discussion_r1100820582

**Comment by justinchuby** [comment]
We can also use `check_function()`

**Comment by BowenBao** [comment]
> We can also use `check_function()`

+1 if https://github.com/onnx/onnx/issues/4872 can be done.
